lib/seattleflu/db/cli/command/etl/longitudinal.py

```diff
@@ -0,0 +1,303 @@
+
+        else:
+            LOG.info("Rolling back all changes; the database will not be modified")
+            db.rollback()
+
+
+def site_identifier(site_name: str) -> str:
+    """
+    Given a *site_name*, returns its matching site identifier.
+    """
+    site_map = {
+        1: "HutchKids",
+        2: "WestCampusChildCareCenter"
+    }
+    if site_name not in site_map:
+        raise UnknownSiteError(f"Unknown site name «{site_name}»")
+
+    return site_map[site_name]
+
+
+def encounter_details(document: dict) -> dict:
+    """
+    Describe encounter details in a simple data structure designed to be used
+    from SQL.
+    """
+    return {
+            "age": age_to_delete(document.get("age")), # XXX TODO: Remove age from details
+            "locations": {
+                "home": {
+                    "region": document.get("census_tract"),
+                }
+            },
+            "responses": {
+                "Race": race(document.get("Race")),
+                "FluShot": flu_shot(document.get("FluShot")),
+                "AssignedSex": [sex(document.get("AssignedSex"))],
+                "HispanicLatino": hispanic_latino(document.get("HispanicLatino")),
+                "MedicalInsurance": insurance(document.get("MedicalInsurance"))
+            },
+        }
+
+
+def sex(sex_name) -> str:
+    """
+    Given a *sex_name*, returns its matching sex identifier.
+    """
+    sex_map = {
+        1: "male",
+        2: "female"
+    }
+
+    return sex_map.get(sex_name, "other")
+
+
+def race(races: list) -> list:
+    """
+    Given a *race_name*, returns the matching race identifier found in Audere
+    survey data. Converts multiple races to 'other'.
+    """
+    if races is None:
+        LOG.debug("No race response found.")
+        return [None]
+
+    if len(races) > 1:
+        return 'other'
+
+    race_map = {
+        "native_american": "americanIndianOrAlaskaNative",
+        "asian": "asian",
+        "black": "blackOrAfricanAmerican",
+        "native_hawaiian": "nativeHawaiian",
+        "white": "white",
+        "other": "other",
+        "unknown": None,
+    }
+
+    race_name = races[0]
+    if race_name not in race_map:
+        raise UnknownRaceError(f"Unknown race name «{race_name}»")
+
+    return [race_map[race_name]]
+
+
+def hispanic_latino(ethnic_group: str) -> list:
+    """
+    Given an *ethnic_group*, returns yes/no value for HispanicLatino key.
+    """
+    if ethnic_group is None:
+        LOG.debug("No ethnic group response found.")
+        return [None]
+
+    ethnic_map = {
+        1: "yes",
+        2: "no",
+        3: None,
+    }
+
+    if ethnic_group not in ethnic_map:
+        raise UnknownEthnicGroupError(f"Unknown ethnic group «{ethnic_group}»")
+
+    return [ethnic_map[ethnic_group]]
+
+
+def flu_shot(flu_shot_response: str) -> list:
+    """
+    Given a *flu_shot_response*, returns a 'yes', 'no', or 'doNotKnow' value for
+    FluShot key.
+    """
```

Comment:
FYI  I noticed that we don't have consistent coding for some of these variables. For example, values for both 'doNotKnow' and 'dontKnow' exist in the db.

Reply:
Yeah, this is because we haven't modeled vaccinations yet in the database but are just using whatever Audere handed us in the enrollment document.

We'll address this inconsistency when we start handling vaccinations.